packages/block-library/src/search/edit.native.js

```diff
@@ -155,21 +157,35 @@ export default function SearchEdit( {
 	);
 
 	const mergeWithBorderStyle = ( style ) => {
-		return { ...style, ...styles.border };
+		const baseStyle = getStylesFromColorScheme(
+			styles.border,
+			styles.borderDark
+		);
+		return { ...style, ...baseStyle };
 	};
 
 	const renderTextField = () => {
-		const inputStyle =
-			buttonPosition === 'button-inside'
-				? styles.searchTextInput
-				: mergeWithBorderStyle( styles.searchTextInput );
+		const getInputStyle = () => {
+			const baseStyle = getStylesFromColorScheme(
+				styles.plainTextInput,
+				styles.plainTextInputDark
+			);
+			return buttonPosition === 'button-inside'
+				? baseStyle
+				: mergeWithBorderStyle( baseStyle );
+		};
```

Comment:
Then if we made the changes from the suggestion above, we could easily merge styles using this approach:

```suggestion
		const inputStyles = [
			getStylesFromColorScheme(
				styles.plainTextInput,
				styles.plainTextInputDark
			),
			! isButtonInside && borderStyle,
		];
```

What do you think?

Reply:
And for the `searchBarStyle` case something similar:

```javascript
	const searchBarStyle = [
		styles.searchBarContainer,
		isButtonInside && borderStyle,
		isButtonInside && styles.searchBarContainer,
	];
```

Reply:
Nice! Made suggested changes in 1b17b727a80a29459f4c66790c531c341c7b6825